### PR TITLE
Re-recognize operator in Help-me-carry

### PR DESCRIPTION
**Pull request by LoyVanBeek**

### CODE REVIEW

**Comment by LoyVanBeek** [comment]
Needs more work:

```
[INFO] [1496174362.371057]: 'Gotcha! I will follow you!'
NOW!!!
Operator already lost. Getting closest possible person entity at 1.5 m in front, radius = 1
Checking end criteria
Operator is at 0.720756 meters distance
Checking if done following
apparently not
Operator is standing still for 0.000010 seconds
Operator within self._lookat_radius
[ERROR] [1496174365.866799623]: LPI: Failed to get robot orientation constraint frame: ED returns 'no such entity'. (f034d9fb427e624c5e9528a5a412e71e-laser).
[ERROR] [1496174365.876855487]: LPI: Failed to get robot orientation constraint frame: ED returns 'no such entity'. (f034d9fb427e624c5e9528a5a412e71e-laser).
[INFO] [1496174366.441260]: 'Stop! I lost you! Until I find you again, please wait there.'
Checking end criteria
Operator is at 0.720756 meters distance
Robot dit not move for x seconds: 0.890092
Robot dit not move for x seconds: 0.945128
Checking if done following
apparently not
Operator within self._lookat_radius
Checking end criteria
Operator is at 0.720756 meters distance
Robot dit not move for x seconds: 1.687951
lost operator and within lookat radius and standing still for 1 second
[INFO] [1496174367.438009]: Trying to recover the operator
[INFO] [1496174367.438617]: 'operator, please look at me while I am looking for you'
[INFO] [1496174368.669635]: getting one image...
[INFO] [1496174368.701376]: got 3932160 bytes of image data
[INFO] [1496174370.005077]: found 1 face(s) in the image
[INFO] [1496174370.005309]: _recover_operator: raw_detections: [categorical_distribution: 
  probabilities: 
    - 
      label: operator
      probability: 5.62451601028
    - 
      label: operator
      probability: 4.5912604332
    - 
      label: operator
      probability: 2.59782600403
    - 
      label: operator
      probability: 2.59451842308
    - 
      label: operator
      probability: 1.84155607224
  unknown_probability: 0.0
roi: 
  x_offset: 489
  y_offset: 264
  height: 260
  width: 222
  do_rectify: False]
[INFO] [1496174370.022760]: project_rois response: points: 
  - 
    header: 
      seq: 0
      stamp: 
        secs: 1496173600
        nsecs: 371457815
      frame_id: /amigo/top_kinect/openni_rgb_optical_frame
    point: 
      x: -0.14816689909
      y: -0.246335408313
      z: 4.09300041199
Transforming roi to map
Could not find an entity 0.8 meter near [      2.0618,     6.33594,     2.28832] @ map
[INFO] [1496174370.850291]: getting one image...
[INFO] [1496174370.964037]: got 3932160 bytes of image data
[INFO] [1496174372.218304]: found 0 face(s) in the image
[INFO] [1496174372.218492]: _recover_operator: raw_detections: []
[INFO] [1496174372.919952]: getting one image...
[INFO] [1496174372.984168]: got 3932160 bytes of image data
[INFO] [1496174374.162480]: found 0 face(s) in the image
[INFO] [1496174374.162607]: _recover_operator: raw_detections: []
[INFO] [1496174375.064460]: getting one image...
[INFO] [1496174375.128087]: got 3932160 bytes of image data
[INFO] [1496174376.275379]: found 0 face(s) in the image
[INFO] [1496174376.275540]: _recover_operator: raw_detections: []
[INFO] [1496174377.477748]: getting one image...
[INFO] [1496174377.509331]: got 3932160 bytes of image data
[INFO] [1496174378.753949]: found 1 face(s) in the image
[INFO] [1496174378.754364]: _recover_operator: raw_detections: [categorical_distribution: 
  probabilities: 
    - 
      label: operator
      probability: 8.7637796402
    - 
      label: operator
      probability: 7.09065580368
    - 
      label: operator
      probability: 5.48053741455
    - 
      label: operator
      probability: 3.7559466362
    - 
      label: operator
      probability: 2.88500547409
  unknown_probability: 0.0
roi: 
  x_offset: 510
  y_offset: 243
  height: 260
  width: 221
  do_rectify: False]
[INFO] [1496174378.789729]: project_rois response: points: 
  - 
    header: 
      seq: 0
      stamp: 
        secs: 1496173600
        nsecs: 371457815
      frame_id: /amigo/top_kinect/openni_rgb_optical_frame
    point: 
      x: -0.0721838739159
      y: -0.324808667628
      z: 4.09300041199
Transforming roi to map
Could not find an entity 0.8 meter near [     2.17784,     6.31869,     2.36611] @ map
[INFO] [1496174379.606608]: getting one image...
[INFO] [1496174379.638202]: got 3932160 bytes of image data
[INFO] [1496174380.906551]: found 0 face(s) in the image
[INFO] [1496174380.906722]: _recover_operator: raw_detections: []
[INFO] [1496174381.608246]: getting one image...
[INFO] [1496174381.722051]: got 3932160 bytes of image data
[INFO] [1496174382.890524]: found 0 face(s) in the image
[INFO] [1496174382.890656]: _recover_operator: raw_detections: []
...
[INFO] [1496174427.519907]: project_rois response: points: 
  - 
    header: 
      seq: 0
      stamp: 
        secs: 1496173600
        nsecs: 371457815
      frame_id: /amigo/top_kinect/openni_rgb_optical_frame
    point: 
      x: 1.26565013082
      y: 0.16935760423
      z: 3.90700030327
Transforming roi to map
Could not find an entity 0.8 meter near [    0.319251,     6.05132,     1.85385] @ map
Operator within self._lookat_radius
[INFO] [1496174429.716062]: 'I am unable to recover you'
[INFO] [1496174431.658029]: State machine transitioning 'FOLLOW_OPERATOR':'lost_operator'-->'ASK_FOLLOW_OR_REMEMBER'
[INFO] [1496174431.658301]: 'Are we at the car or should I follow you?'
```

**Comment by MatthijsBurgh** [comment]
Does this one still need to be merged? Because #295 is already merged?

**Comment by LoyVanBeek** [comment]
This eventually still needs to be merged, these fixes all apply to different issues with challenge_help_me_carry. 

I'll change the name accordingly.

File: robot_skills/src/robot_skills/head.py [hunk 1]
```diff
@@ -306,18 +315,75 @@ def learn_person(self, name='operator'):
...
+        :returns image_recognition_msgs/Recognition
+        """
         image = self.get_image()
         if stamp:
             return self._get_faces(image).recognitions, image.header.stamp
         else:
             return self._get_faces(image).recognitions
 
+    def get_best_face_recognition(self, recognitions, desired_label):
+        """Returns the Recognition with the highest probability of having some label.
```
Comment (Rayman) [inline robot_skills/src/robot_skills/head.py]: of heaving *the* desired label

**Pull request merged**